src/controllers/update/secure.update.filter.controller.ts

```diff
@@ -0,0 +1,41 @@
+import { NextFunction, Request, Response } from "express";
+
+import * as config from "../../config";
+import { logger } from "../../utils/logger";
+import { ApplicationData } from "../../model";
+import { getApplicationData, setExtraData } from "../../utils/application.data";
+import { IsSecureRegisterKey } from "../../model/data.types.model";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `GET ${config.SECURE_REGISTER_FILTER_PAGE}`);
+    const appData: ApplicationData = getApplicationData(req.session);
+    const notLiveTitle: string = "(NOT LIVE)";
+    return res.render(config.SECURE_REGISTER_FILTER_PAGE, {
+      backLinkUrl: config.UPDATE_LANDING_PAGE_URL,
+      templateName: config.SECURE_REGISTER_FILTER_PAGE,
+      [IsSecureRegisterKey]: appData[IsSecureRegisterKey],
+      notLiveTitle
+    });
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+export const post = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `POST ${config.SECURE_REGISTER_FILTER_PAGE}`);
+    const isSecureRegister = req.body[IsSecureRegisterKey];
+
+    setExtraData(req.session, { ...getApplicationData(req.session), [IsSecureRegisterKey]: isSecureRegister });
+
+    if (isSecureRegister === '0') {
+      return res.redirect(config.OVERSEAS_ENTITY_QUERY_PAGE);
+    }
```

Comment:
What happens in case 0? In registration we end up on a secure register update card.

Reply:
We decided that it will be covered in UAR-89 as it requires that page. Case 1 is when you select the `Yes` radio button.

Reply:
Originally, I had case 1 in there as well but it was just an empty condition as it could not redirect the user to anything until the UAR-89 apply paper form page existed.